lib/linguist/heuristics.rb

```diff
@@ -160,6 +160,20 @@ def call(data)
       end
     end
 
+    disambiguate "M", "Mathematica", "Matlab", "Mercury", "Objective-C" do |data|
+      if /^[ \t]*@(interface|class|protocol|property|end|synchronised|selector|implementation)\b/.match(data)
```

Comment:
Makes sense. I've just copied the regex from here: https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb#L81